lib/logstash/filters/grok.rb

```diff
@@ -180,6 +180,10 @@ class LogStash::Filters::Grok < LogStash::Filters::Base
     #     NUMBER \d+
     config :patterns_dir, :validate => :array, :default => []
 
+    # Glob pattern, used to select the pattern files in the directories
+    # specified by patterns_dir
+    config :patterns_files_glob, :validate => :string, :default => "*"
```

Comment:
It feels weird to have two settings to specify the files to load. While I see this new setting as needed for backwards compatibility, I wonder if we could add glob support to patterns_dir instead, and later renaming it to `patterns_path` -- thoughts on this approach?

(I'm open to deferring the patterns_path stuff to another time outside this PR)